File: src/main/java/com/moovie/backend/domain/usecase/movies/GetGenres.java
package com.moovie.backend.domain.usecase.movies;

import com.moovie.backend.domain.model.Genre;
import com.moovie.backend.domain.repository.MoviesRepository;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class GetGenres {

    private final MoviesRepository moviesRepository;

    public GetGenres(MoviesRepository moviesRepository) {
        this.moviesRepository = moviesRepository;
    }

    public List<Genre> invoke() {
        return moviesRepository.getGenres();
    }
}

File: src/main/kotlin/org/example/domain/usecase/movies/GetGenres.kt
package org.example.domain.usecase.movies

import kotlinx.coroutines.runBlocking
import org.example.domain.model.Genre
import org.example.domain.repository.MoviesRepository

class GetGenres(private val repository: MoviesRepository) {
    operator fun invoke(): List<Genre> = runBlocking {
        repository.getGenres()
    }
}
